contentctl/actions/detection_testing/infrastructures/DetectionTestingInfrastructure.py

```diff
@@ -60,13 +60,14 @@ class CleanupTestGroupResults(BaseModel):
 
 class ContainerStoppedException(Exception):
     pass
+class CannotRunBaselineException(Exception):
```

Comment:
Yes, I have given a descriptive docstring and done a bit of reorganization here:
https://github.com/splunk/contentctl/pull/260/files#diff-701b2a9b8fd8a8d35960dcf412eb3a87f0639121004f98cb277832179df81faeR666-R668

Reply:
I have done a few things.  First, as suggested, I added some documentation.  Second, I updated the detection to "attempt" to run a Baseline and throw a runtime error if the search has any Baselines: https://github.com/splunk/contentctl/pull/260/files#diff-701b2a9b8fd8a8d35960dcf412eb3a87f0639121004f98cb277832179df81faeR666-R671
I would like this to be a test runtime error, rather than a validation error, because I want us to be able to build apps with Baseline searches (when detections are not marked manual_test for now. 

Forcing us to mark all of these manual_test reduces the chance we will revisit them, so making these runtime errors will ensure they will be revisited.

I have also removed baselines from the TEST objects because they are part of the detection, not the test:
https://github.com/splunk/contentctl/pull/260/files#diff-f5f25273cc9fbc9cdc9a96d6c30ed166bd825d73644c366226b94b130e6e7b67L24-L26

Reply:
So does this mean we will be seeing these detections fail now when we run a full package test?

Reply:
Right now ALL of the detections which include baselines AND are `production` are marked as `manual_test` (so they were being skipped). As such, this should not result in any additional failures for our current content. 
But if I build a new search using a baseline that is NOT marked as `manual_test`, it will throw a baseline error that looks like
```yml
summary:
  mode: changes
  enable_integration_testing: false
  success: false
  total_detections: 1
  total_tested_detections: 1
  total_pass: 0
  total_fail: 1
  total_skipped: 0
  total_untested: 0
  total_production: 1
  total_experimental: 0
  total_deprecated: 0
  total_manual: 0
  total_other_skips: 0
  success_rate: 0.0%
tested_detections:
- name: Cloud Provisioning Activity From Previously Unseen City
  type: Anomaly
  production_status: production
  status: fail
  source_category: cloud
  data_source:
  - AWS CloudTrail
  search: '| tstats earliest(_time) as firstTime, latest(_time) as lastTime from datamodel=Change
    where (All_Changes.action=started OR All_Changes.action=created) All_Changes.status=success
    by All_Changes.src, All_Changes.user, All_Changes.object, All_Changes.command
    | `drop_dm_object_name("All_Changes")` | iplocation src | where isnotnull(City)
    | lookup previously_seen_cloud_provisioning_activity_sources City as City OUTPUT
    firstTimeSeen, enough_data | eventstats max(enough_data) as enough_data | where
    enough_data=1 | eval firstTimeSeenCity=min(firstTimeSeen) | where isnull(firstTimeSeenCity)
    OR firstTimeSeenCity > relative_time(now(), `previously_unseen_cloud_provisioning_activity_window`)
    | `security_content_ctime(firstTime)` | table firstTime, src, City, user, object,
    command | `cloud_provisioning_activity_from_previously_unseen_city_filter`'
  file_path: detections/cloud/cloud_provisioning_activity_from_previously_unseen_city.yml
  manual_test: null
  success: false
  tests:
  - name: True Positive Test
    test_type: unit
    success: false
    message: 'EXCEPTION: Detection requires Execution of a Baseline, however Baseline
      execution is not currently supported in contentctl. Mark this as manual_test.'
    exception: Detection requires Execution of a Baseline, however Baseline execution
      is not currently supported in contentctl. Mark this as manual_test.
    status: error
    duration: 6.28
    wait_duration: null
    resultCount: null
    runDuration: null
  - name: True Positive Test
    test_type: integration
    success: false
    message: 'TEST FAILED (PREEMPTIVE): associated unit test failed or encountered
      an error'
    exception: Detection requires Execution of a Baseline, however Baseline execution
      is not currently supported in contentctl. Mark this as manual_test.
    status: error
    duration: 5.33
    wait_duration: null
    resultCount: null
    runDuration: null
skipped_detections: []
untested_detections: []
percent_complete: UKNOWN
```

Reply:
Ok cool! Ty :)